### ISSUE TITLE
VideoEffect is applied in duplicate.

### ISSUE THREAD

**Issue by HyunjoonKo** [shogo4405/HaishinKit.swift#904]
When implementing and applying a custom `VideoEffect`, we found that the color of the preview displayed on the sending device and the video received on the receiving device are different.

A custom `VideoEffect` implemented a LUT filter (HALD8) with [Brightroom](https://github.com/muukii/Brightroom) as follows:

```swift
import UIKit
import HaishinKit
import Brightroom
import AVFoundation

final class Cubeffect: VideoEffect {
    var filter: FilterColorCube? = nil

    init(effect: FilterColorCube) {
        self.filter = effect
    }
    
    override func execute(_ image: CIImage, info: CMSampleBuffer?) -> CIImage {
        guard let filter = filter else {
            return image
        }
        
        let modifiedImage = filter.apply(to: image, sourceImage: image)
        return modifiedImage
    }
}
```
Then, the color of the video looked different on the sending device and the receiving device.

Sender (original, no effect):
![IMG_E28584FF3CA0-1](https://user-images.githubusercontent.com/3861180/132453250-841291f3-cdd2-4a9f-b19b-74d07d69617c.jpeg)

Sender (using Cubeffect):
![IMG_0407](https://user-images.githubusercontent.com/3861180/132450067-acd3a4ab-6036-4467-88d5-4f0f94c6b94a.PNG)

Receiver (Use Cubeffect on the sender side):
![IMG_0409_R](https://user-images.githubusercontent.com/3861180/132455601-a516e052-1aef-46dd-9e21-bcc046e9cb04.jpg)

I found that the VideoEffect on the device sending the video was too much.
Looking at the cause, there was a problem with `encodeSampleBuffer(_:)` of `VideoIOComponent`.

In conclusion, the filtered image should be used for `context?.render(image, to: imageBuffer ?? buffer)` and the original image should be used for `renderer?.render(image: image)`. Because `let image: CIImage = effect(buffer, info: sampleBuffer)` already returns the image with VideoEffect applied.

So I modified it like this:
```swift
renderer?.render(image: CIImage(cvPixelBuffer: buffer))
```
In this case, VideoEffect was applied to both the device sending and receiving the video. Can you please review it?

**Smartphone (please complete the following information):**
 - Device: iPhone6, iPhone XS, iPhone 12 pro
 - OS: iOS 12.5.2, 14.5
 - Version 1.1.6

**Comment by HyunjoonKo** [comment]
This issue seems to be resolved in version 1.2.0.
Close the issue.
